### ISSUE TITLE
Investigate using conda yaml file for setting up environment

### ISSUE THREAD

**Issue by laurejt** [Princeton-CDH/ppa-django#585]

**Comment by laurejt** [comment]
**TLDR:** Conda's current yaml support for environment management is not a good fit for the current workflow (i.e., multiple requirements files for different situations).

Conda environments can be [created](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file), [updated](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#updating-an-environment), and [shared](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#exporting-an-environment-file-across-platforms) via yaml files. However, this generally only works with a *single* yaml file. There are generally two kinds of issues when using multiple files:

1. **Updates lose track of pinned dependencies.** Although the `conda env update -f [env.yml]` command can update the current environment according to a yaml file, pinned dependencies can be updated. The third-party tool [`conda-merge`](https://github.com/amitbeka/conda-merge) might be able to help, but it further complicates the workflow and may still not produce a working environment (see the second issue).

2. **Conda and pip don't always play nice.** The interactivity / interoperability of pip within conda environments has changed over the years, but the results are still mixed (see: [[1]](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#using-pip-in-an-environment), [[2]](https://conda.discourse.group/t/defining-and-documenting-how-pip-should-interact-with-conda-environments/200). Generally, once packages have been installed using pip, it shouldn't be updated further using conda.

Anecdotally, I've generally had a great experience using yml files for conda environment creation/management. However, this only works when there's a *single* file, not multiple.
